### PR TITLE
Wrapper: use generics

### LINKED ISSUE #55: idea: use generics in type_wrapper.go
after looking at code I got an idea, to use `go 1.19`'s generic to replace `wrap*` with one large generic method

### PR DESCRIPTION
**Pull request by gucio321**
fix #55

### CODE REVIEW

**Comment by gucio321** [comment]
I'm going to do the same for slice types

**Comment by gucio321** [comment]
> I'm going to do the same for slice types

well nevermind - it is not used the way I tought it is :smile:

**Comment by eliasdaler** [comment]
Please don't use PR's for drafts - develop in your own fork. It pollutes notifications - I'm only interested in final code.

**Comment by gucio321** [comment]
it is already ready for review

File: type_wrapper.go [hunk 1]
```diff
@@ -29,29 +29,27 @@ func wrapBool(goValue *bool) (wrapped *C.bool, finisher func()) {
...
+// because it can be directly converted to int)
+type Number interface {
+	~int8 | ~int16 | ~int32 | ~int64 |
+		~uint8 | ~uint16 | ~uint32 | ~uint64 |
+		~float32 | ~float64
 }
 
-func wrapFloat(goValue *float32) (wrapped *C.float, finisher func()) {
+// wrapPtrCType is a generic method to convert GOTYPE (int32/float32 e.t.c.) into CTYPE (c_int/c_float e.t.c.)
+func wrapPtrCType[CTYPE Number, GOTYPE Number](goValue *GOTYPE) (wrapped *CTYPE, finisher func()) {
```
Comment (eliasdaler) [inline type_wrapper.go:42]: I'd also just name it `wrapNumberPtr` because it can only wrap numbers - it's not fully generic.

**Review by eliasdaler — CHANGES_REQUESTED** [review]

File: type_wrapper.go (hunk 1 above)
Comment (gucio321) [reply type_wrapper.go:42]: ok, good idea

**Review by eliasdaler — APPROVED** [review]

**Pull request merged**